Tests/JWTTests/JWTTests.swift

```diff
@@ -100,14 +112,48 @@ final class JWTTests: XCTestCase {
         try jwt.verifySignature(using: signer)
     }
 
-    static let all = [
-        ("testSignature", testSignature),
-        ("testInitWithToken", testInitWithToken),
-        ("testIncorrectNumberOfSegments", testIncorrectNumberOfSegments),
-        ("testDefaultHeaders", testDefaultHeaders),
-        ("testCustomHeaders", testCustomHeaders),
-        ("testCustomJSONHeaders", testCustomJSONHeaders),
-        ("testJWTClaimsCanBeVerified", testJWTClaimsCanBeVerified),
-        ("testHS256VerificationOfWellKnownToken", testHS256VerificationOfWellKnownToken)
-    ]
+    func testHeaders() throws {
+        let expiry = Date() + 1800
+        let headers: JSON = JSON(["aaaa": "bbbb", "cccc": 1])
+        let payload = JSON([
+            ExpirationTimeClaim(date: expiry)
+        ])
+        let fail = try JWT(headers: headers,payload: payload, signer: HS512(key: "secret".bytes))
+        let token = try fail.createToken()
+
+
+        let receivedJWT = try JWT(token: token)
+        do {
```

Comment:
If you rely on the thrown exception, the failure messages will be "JWTError.Error 0" or similar. However, the Swift description of the error (which is printed and manually passed as the failure message) is much more descriptive.